### ISSUE TITLE
Default Canvas stroke width is 2; should be 1

### ISSUE THREAD

**Issue by mhsmith** [beeware/toga#2205]
### Describe the bug

Toga's default stroke width is 2, but the [HTML spec says](https://html.spec.whatwg.org/multipage/canvas.html#line-styles) the width "must initially have the value 1.0"

I'm not sure if this was done to avoid 1-pixel horizontal and vertical lines being spread across 2 pixels, because integer coordinates are treated as pixel edges rather than centers. But the HTML canvas does the same, and its behaviour is [well-known](https://stackoverflow.com/a/7531540). So it's probably better if we just match it.

### Environment

- Software versions:
  - Toga: main (536a60f)

**Comment by HalfWhitt** [comment]
If anyone would like to pick this up:

- This is a pretty minor change, but will have to be made in every canvas backend (backend_name/widgets/canvas.py).
- This will also break a lot of tests in the testbed — any of them that perform a stroke without a custom width set.
   - It might be appropriate to simply alter some tests to use a width of 2, to match the already extant reference images.
   - Otherwise — when canvas tests fail on GitHub CI, the actual resulting image can be downloaded from the test summary page. They'll be zip files, named with the platform name and "app failure data". The images inside can be copied into testbed/src/testbed/resources/canvas/ to replace the old reference images that were generated with 1-pixel default width.

**Comment by HalfWhitt** [comment]
This information has been reiterated in #4381.

**Comment by mhsmith** [comment]
Why did you create a duplicate when you'd already discovered this issue?

**Comment by HalfWhitt** [comment]
I didn't at first, but @kattni pointed out that's the way our [triage process](https://github.com/beeware/beeware/discussions/618) is set up now. That way a contributor, especially a new one, finds all the info about what they need to do summarized in the top post.

**Comment by mhsmith** [comment]
I don't think that was necessary here, as there wasn't any significant discussion to summarize. If you want to add things to the top post, you can edit it.

**Comment by kattni** [comment]
My default is not to assume that I should be editing someone else's issue message. That's why we're doing what we're doing. I understand you're ok with it here, but that might not carry to other issues.

**Comment by mhsmith** [comment]
That's fair enough, but I don't think it's necessary to strictly keep all the information in the top post. If someone makes a  comment on the second issue, are we going to close it and create a third one? As long as the discussion is brief enough for a contributor to easily understand, I think it can be left as is.

**Comment by kattni** [comment]
@mhsmith If further discussion is had on the second ticket, that changes the nature of the initial response, I would expect that we would update the primary message in the issue to reflect those changes. 

The issue I'm trying to address here is that if we let the action items end up spread out across a thread, things will get missed, and we will make more work for ourselves. That is why there are several categories available between "triage" and "ready", such as "needs design" or "under investigation". The idea is that we have the discussion on the first issue, and then generate the second issue when we are certain of the path forward. If that path changes, the initial message should be changed. 

That said, if the original issue was filed by someone who is either able to update the initial message or willing to allow us to modify it for them, then maintaining the initial issue would be acceptable. The whole point of this is to ensure that when an issue is marked as "Ready", the top message contains everything that needs to be done and all expectations for those actions. I am happy to discuss options for how to make this happen, but that is the necessary goal.

**Comment by freakboy3742** [comment]
There's two very valid but distinct issues here, IMHO.

On the one hand, I share @kattni's concern about appearing that we're editing someone else's words. 

On the other hand, I also share @mhsmith's concern about a proliferation of tickets. Opening a new ticket every time makes the history a little harder to navigate, which isn't ideal. 

After an in-person chat with @kattni today, here's a suggestion for how to reconcile these two positions.

This is, ultimately, BeeWare's ticket tracker. As a team, we're trying to curate a list of actionable work for contributors, and we're collectively responsible for that. To that end, we shouldn't be overly worried about editing each others work - we do that in the code all the time, and the Github interface exposes edit history to show earlier forms of a ticket if there's concern about specific attribution. 

To that end, if a ticket is opened by a core team member (as in this case), editing the existing ticket to make it more consumable by contributors should be considered fair game. If a ticket update *does* misrepresent or alter the intention of the original, we're a team, and we can resolve the discrepancy as a team.

However, if a ticket has been created by a community member, we fall back creating a new issue (unless the issue is essentially good as-is, or only needs trivial edits - but history shows that's a pretty rare outcome). Once it's a "team created" issue, we can collectively edit it. 

Essentially, what we end up with is 2 ticket trackers - one for the public to put things into, and one that we use as the "clean" triaged list. It's implemented on one *actual* tracker - but the intended end point is a list of clean, actionable tickets.

Also - we're only creating a new ticket when we're looking at a ticket with a clear, actionable outcome as a result of the triage process. Something like a feature enhancement where there's still discussion happening about the value of the feature or potential approaches would stay on the original ticket until such time as a clear and actionable plan emerges.

@kattni @mhsmith Does that address your concerns?

**Comment by kattni** [comment]
Yes. This is exactly what I had in mind.

**Comment by mhsmith** [comment]
> To that end, if a ticket is opened by a core team member (as in this case), editing the existing ticket to make it more consumable by contributors should be considered fair game. If a ticket update _does_ misrepresent or alter the intention of the original, we're a team, and we can resolve the discrepancy as a team.

Yes, that makes sense. But since edits don't generate notifications, if there's any doubt at all about the edit, we should suggest it as a comment first, or discuss it elsewhere. Then after the comments have been incorporated into the top post, we can hide them.

HalfWhitt's first comment above is an example of one that I'd be fine to be added as an edit without discussion. However, in such cases you can still generate a notification without leaving any clutter, by posting a comment, copying it into the top post, and then deleting it. For example, I did that with this paragraph, which was originally a second comment below this one.
